src/fal_partition.c

```diff
@@ -411,7 +411,7 @@ int fal_partition_read(const struct fal_partition *part, uint32_t addr, uint8_t
     assert(part);
     assert(buf);
 
-    if (addr + size > part->len)
+    if (addr >= part->len || part->len - addr < size)
```

Comment:
```suggestion
    if (addr >= part->len || addr + size > part->len)
```
改成这种方式吧，上面那个感觉逻辑不太直观

Reply:
他的意思是 addr+size 有可能会溢出吧，所以改成了减

Reply:
有测试的用例吗？

Reply:
其实只有当，fal 管理的地址空间达到 4G 的时候，才会出现这个问题。
![021482fd51f8bee487cb0850fe3e4a1](https://user-images.githubusercontent.com/25633294/128796994-37757852-d094-4a46-b5a9-cec676e591c9.png)